x/group/keeper/msg_server.go

```diff
@@ -774,9 +777,15 @@ func (k Keeper) doAuthenticated(ctx sdk.Context, req authNGroupReq, action actio
 
 // assertMetadataLength returns an error if given metadata length
 // is greater than a fixed maxMetadataLength.
-func assertMetadataLength(metadata []byte, description string) error {
-	if len(metadata) > group.MaxMetadataLength {
-		return sdkerrors.Wrap(errors.ErrMaxLimit, description)
+func (k Keeper) assertMetadataLength(metadata []byte, description string) error {
+	if metadata != nil && k.maxMetadataLength != 0 {
+		if uint64(len(metadata)) > k.maxMetadataLength {
+			return sdkerrors.Wrapf(errors.ErrMaxLimit, description)
+		}
```

Comment:
As Marie pointed out, k.maxMetadataLength can never be 0, so this is enough:

```suggestion
	if metadata != nil && uint64(len(metadata)) > k.maxMetadataLength {
		return sdkerrors.Wrapf(errors.ErrMaxLimit, description)
	}
```